### ISSUE TITLE
Control the Order of pipeline deploy sequence

### ISSUE THREAD

**Issue by ampiy** [aws/copilot-cli#3519]
Is there a way to control the deploy sequence in a pipeline, similar docker-compose depends on?

I have around 11 microservices with a nginx gateway service. Everytime the pipeline is triggered, nginx service is deployed without waiting for other services.
This causes the nginx to not have latest local service mapping of the services that were deployed after it. And results in a gateway time out error for these other microservices. 

The only solution is to force redeploy the nginx service. 

Any help on getting around it.

**Comment by efekarakus** [comment]
Hi @ampiy !

We just merged this feature 😄 here is the design proposal that we implemented: https://github.com/aws/copilot-cli/issues/3402
It'll go out next release with an ETA of next week

So from now in your pipeline manifest you'll be able to write the following:
```yaml
stages:
- name: prod
  deployments:
    service1:
    service2:
    service3:
    nginx:
       depends_on: [service1, service2, service3]
```

Would that satisfy your usecase?

**Comment by ampiy** [comment]
> 

That would be perfect.

**Comment by huanjani** [comment]
@ampiy: The feature is now released in v1.18.0: https://github.com/aws/copilot-cli/releases/tag/v1.18.0!
